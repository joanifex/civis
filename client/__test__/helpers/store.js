import configureMockStore from 'redux-mock-store';


const store = (model) => {
  const mockStore = configureMockStore();
  return mockStore(model)
}

export default store;
