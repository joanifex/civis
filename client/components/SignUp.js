import React from 'react';
import { Link } from 'react-router';
import { handleSignUp } from '../actions/auth';
import { connect } from 'react-redux';

class SignUp extends React.Component {
  handleSubmit = (e) => {
    e.preventDefault();
    let email = this.refs.email.value;
    let password = this.refs.password.value;
    let first_name = this.refs.firstName.value;
    let last_name = this.refs.lastName.value;
    this.props.dispatch(handleSignUp(email, password, first_name, last_name));
  }

  render() {
    return(

      <div className="row">
         <div className="col s12 m10 offset-m1 l8 offset-l2">
           <div className="card grey lighten-4">
             <div className="card-content black-text">
               <span className="card-title">Sign Up For A New Account</span>
               <form onSubmit={ this.handleSubmit }>
                 <div className="row">
                   <div className="col s12">
                     <input ref='firstName' type="text" required placeholder="First Name"/>
                     <br />
                     <input ref='lastName' type="text" required placeholder="Last Name"/>
                     <br />
                     <i className="fa fa-envelope-o prefix" aria-hidden="true"></i>
                     <input ref='email' className="icon_prefix" type='text' required placeholder='Email' />
                     <br />
                    <input ref='password' type='password' required placeholder='Password' />
                    <br />
                    <input type='submit' className='btn' value='Sign Up' />
                    <Link to='/login' className='btn grey'>Cancel</Link>
                 </div>
               </div>
               </form>
             </div>
           </div>
         </div>
       </div>
    );
  }
}

export default connect()(SignUp);
