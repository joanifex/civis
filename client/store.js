import { createStore, compose, applyMiddleware } from 'redux';
import { syncHistoryWithStore } from 'react-router-redux';
import { browserHistory } from 'react-router';
import thunk from 'redux-thunk';

import rootReducer from './reducers/index';

const enhancers = compose(
  applyMiddleware(thunk),
  window.devToolsExtension ? window.devToolsExtension() : f => f
)

let appDataSet = document.getElementById('app').dataset;
let remember = appDataSet.remember;

remember = remember === 'true' ? true : false;

const store = createStore(rootReducer, { auth: { isAuthenticated: remember } }, enhancers)

if(module.hot) {
  module.hot.accept('./reducers/',() => {
    const nextRootReducer = require('./reducers/index').default;
    store.replaceReducer(nextRootReducer);
  });
}

export const history = syncHistoryWithStore(browserHistory, store)

export default store
