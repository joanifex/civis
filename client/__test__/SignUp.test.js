import React from 'react';
import SignUp from '../components/SignUp.js';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from './helpers/store';

test('Component is rendered', () => {
  const component = renderer.create(
    <Provider store={store({})}>
      <SignUp />
   </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
