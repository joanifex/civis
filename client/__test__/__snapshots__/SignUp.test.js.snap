// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is rendered 1`] = `
<div
  className="row"
>
  <div
    className="col s12 m10 offset-m1 l8 offset-l2"
  >
    <div
      className="card grey lighten-4"
    >
      <div
        className="card-content black-text"
      >
        <span
          className="card-title"
        >
          Sign Up For A New Account
        </span>
        <form
          id="sign_up"
          onSubmit={[Function]}
        >
          <div
            className="row"
          >
            <div
              className="col s12"
            >
              <div
                className="input-field"
              >
                <i
                  className="fa fa-user-circle-o prefix"
                />
                <input
                  className="icon_prefix"
                  placeholder="First Name"
                  required={true}
                  type="text"
                />
              </div>
              <div
                className="input-field"
              >
                <i
                  className="fa fa-user-circle-o prefix"
                />
                <input
                  className="icon_prefix"
                  placeholder="Last Name"
                  required={true}
                  type="text"
                />
              </div>
              <div
                className="input-field"
              >
                <i
                  className="fa fa-envelope-o prefix"
                />
                <input
                  className="icon_prefix"
                  placeholder="Email"
                  required={true}
                  type="email"
                />
              </div>
              <div
                className="input-field"
              >
                <i
                  className="fa fa-unlock-alt prefix"
                />
                <input
                  className="icon_prefix"
                  placeholder="Password"
                  required={true}
                  type="password"
                />
              </div>
              <div
                className="input-field"
              >
                <i
                  className="fa fa-unlock-alt prefix"
                />
                <input
                  className="icon_prefix"
                  placeholder="confirm Password"
                  required={true}
                  type="password"
                />
              </div>
              <input
                className="btn"
                type="submit"
                value="Sign Up"
              />
              <a
                className="btn grey"
                onClick={[Function]}
                style={Object {}}
              >
                Cancel
              </a>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
